### PR TITLE
Removed inconsistency in the `ShortcutManager`

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

At first, I thought that it may be Gecko bug. However, after some research I've found that it seems to be a desired change that `keydown` and `keyup` events are thrown (even when there is no key pressed) for browser's autofill action. There was a bugfix which [introduced that feature in Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=581537).

After further investigation I've found that we have some inconsistency. Events outside the table shouldn't be handled (within the issue is performed event on `input` outside the table). While the `keydown` event wasn't handled the `keyup` has been. Thus, the `KeyUp` event (without `key` property) has been handled.

This PR fixed inconsistency in the `ShortcutManager` and introducing checking whether an `event` has `key` defined.

### How has this been tested?

Checked an example provided in the issue.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9562 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/shortcuts/recorder.js [hunk 1]
```diff
@@ -124,6 +133,10 @@ export function useRecorder(ownerWindow, beforeKeyDown, afterKeyDown, callback)
    * @private
    */
   const onblur = () => {
+    if (handleEvent() === false) {
+      return;
+    }
+
```
Comment (budnix) [inline handsontable/src/shortcuts/recorder.js:139]: This line causes a bug where the Control/Command key is not released when the browser window is blurred. Thus, after the window is focused again the non-contiguous selection is locked even when the meta key is not pressed at all. 

Kapture 2022-06-09 at 08 24 19

**Review by budnix — CHANGES_REQUESTED** [review]

**Review by budnix — APPROVED** [review]
👍

**Comment by wszymanski** [comment]
@warpech @aninde Should it be merged? Will it be released for `12.1.0`?

**Comment by budnix** [comment]
> @warpech @aninde Should it be merged? Will it be released for `12.1.0`?

The issue is marked to be released in 12.1.0 in [the ZenDesk](https://app.zenhub.com/workspaces/handsontable-development-619c0f46f1c2d20014fc4595/issues/handsontable/handsontable/9562). Definitely you should merge it :)

**Comment by wszymanski** [comment]
> > @warpech @aninde Should it be merged? Will it be released for `12.1.0`?
> 
> The issue is marked to be released in 12.1.0 in [the ZenHub](https://app.zenhub.com/workspaces/handsontable-development-619c0f46f1c2d20014fc4595/issues/handsontable/handsontable/9562). Definitely you should merge it :)

Indeed, thanks.

**Pull request merged**